### PR TITLE
[ProgressView] Address progress view example problems with safe area insets and impr…

### LINKED ISSUE #3718: [ProgressView] Demo does not lay out views correctly after rotation, ignores safe area
### Overview
When starting it in Landscape, the notch is overlapping portions of the examples. When rotated to Portrait, the first example is hidden behind the App Bar. When starting in Portrait and rotating to landscape, the content is incorrectly offset from the App Bar and the superview's background can be seen. This results in a black "box" below the App Bar.

**Portrait (rotated)**
simulator screen shot - iphone x - 2018-05-04 at 10 48 41

**Landscape (original)**
screen shot 2018-05-04 at 10 51 34 am

**Landscape (rotated)**
screen shot 2018-05-04 at 10 51 25 am

### Reproduction steps
1. Open the catalog in Portrait
1. Open Progress View &rarr; Start Demo
1. Note the overlap of the notch and the examples
1. Close and open Progress View &rarr; Start Demo
1. Rotate to Portrait
1. Note that the first progress bar is hidden behind the App Bar

### Version number
55.0.1

### Operating system and device
iPhone X/iOS 11.3 (Simulator)

### Related issues
#3705

### LINKED ISSUE #4785: [ProgressView] Demo labels have wrong alignment in landscape mode on iOS12
Progress view labels appear on the right instead of the left on iOS 12 in landscape mode. 
This seems to occur only on iOS12 (tested on: iPhone 8 Plus simulator, iPhoneX simulator & iPhoneX device).
This could potentially be an issue with the progress view component which manifests only on iOS12.

Note: the view being partially covered in iPhoneX devices is tracked by: ["[ProgressView] Demo ignores safe area"](https://github.com/material-components/material-components-ios/issues/3718).

iOS 12 (iPhone X, 8+):
progress view - iphone x - ios12

iOS 11.4 (iPhone X, 8+):
progress view - iphone x - ios11.4

### PR DESCRIPTION
**Pull request by andrewoverton**
I'm not sure what everybody's feelings towards VFL and single character variable names are but I make use of both of them here, for consistency's sake. 

Closes #4785 
Closes #3718

"Before" pic for both issues:
image

"After" pics for both issues: 
simulator screen shot - iphone x - 2018-08-24 at 16 19 50
simulator screen shot - iphone x - 2018-08-24 at 16 19 46

### CODE REVIEW

File: components/ProgressView/examples/ProgressViewExample.m [hunk 1]
```diff
@@ -119,45 +122,57 @@ - (void)viewDidLoad {
   self.navigationItem.rightBarButtonItem.accessibilityIdentifier = @"animate_button";
 }
 
+-(void)setupContainer {
+  self.container = [[UIView alloc] initWithFrame:self.view.bounds];
+  self.container.translatesAutoresizingMaskIntoConstraints = NO;
+  [self.view addSubview:self.container];
+  UILayoutGuide *layoutGuide = self.view.layoutMarginsGuide;
```
Comment (romoore) [inline components/ProgressView/examples/ProgressViewExample.m]: This is iOS 9+, right? I don't think we have dropped iOS 8 yet, but I could easily be mistaken.

File: components/ProgressView/examples/ProgressViewExample.m [hunk 2]
```diff
@@ -170,13 +185,14 @@ - (void)setupConstraints {
     @"backwardAnimateLabel" : _backwardProgressAnimateLabel,
   };
   NSDictionary *metrics = @{
-    @"p" : @20,
+    @"t" : @20,
```
Comment (romoore) [inline components/ProgressView/examples/ProgressViewExample.m:198]: Does this work correctly with the in-call status bar? Is there a way we could work with the `safeAreaInsets` or `safeAreaLayoutGuide` instead?

File: components/ProgressView/examples/ProgressViewExample.m [hunk 3]
```diff
@@ -194,11 +210,11 @@ - (void)setupConstraints {
     @"H:|-(p)-[coloredView]-(p)-|",
     @"H:|-(p)-[backwardResetView]-(p)-|",
     @"H:|-(p)-[backwardAnimateView]-(p)-|",
-    @"H:|-(>=p)-[stockLabel]-(>=p)-|",
-    @"H:|-(>=p)-[tintedLabel]-(>=p)-|",
-    @"H:|-(>=p)-[coloredLabel]-(>=p)-|",
-    @"H:|-(>=p)-[backwardResetLabel]-(>=p)-|",
-    @"H:|-(>=p)-[backwardAnimateLabel]-(>=p)-|",
+    @"H:|-(p)-[stockLabel]-|",
```
Comment (romoore) [inline components/ProgressView/examples/ProgressViewExample.m]: This change will stretch the width of the label, relying on the text alignment for positioning. Is this what you want, or do should we allow the UILabel to hug its content?

**Review by romoore — DISMISSED** [review]
If we still support iOS 8, a different layout choice will be needed for the container view.

File: components/ProgressView/examples/ProgressViewExample.m (hunk 3 above)
Comment (andrewoverton) [reply components/ProgressView/examples/ProgressViewExample.m]: I'm fine with it the way it is. Do you have a preference?

File: components/ProgressView/examples/ProgressViewExample.m (hunk 2 above)
Comment (andrewoverton) [reply components/ProgressView/examples/ProgressViewExample.m:198]: It does work with the status bar. I looked into safeAreaInsets but ended up using `layoutMargins` in ecb2c39. Does that work for you? I tested it again with iPhone 8 in iOS 11 and iPhone X in iOS 11 and 12.

File: components/ProgressView/examples/ProgressViewExample.m [hunk 4]
```diff
@@ -25,6 +25,8 @@
 
 @interface ProgressViewExample : UIViewController
 
+@property(nonatomic, strong) UIView *container;
+
```
Comment (codeman7) [inline components/ProgressView/examples/ProgressViewExample.m:29]: Do we need the extra line?

File: components/ProgressView/examples/ProgressViewExample.m [hunk 5]
```diff
@@ -119,45 +122,67 @@ - (void)viewDidLoad {
...
+  [self positionContainer];
+}
+
+-(void)setupContainer {
+  self.container = [[UIView alloc] initWithFrame:self.view.bounds];
+  [self.view addSubview:self.container];
+}
+
+- (void)positionContainer {
+  CGFloat originX = self.view.bounds.origin.x + self.view.layoutMargins.left;
```
Comment (codeman7) [inline components/ProgressView/examples/ProgressViewExample.m]: I believe we need to use either `CGRectGetMinX(), CGRectGetMinY(), etc` or standardize the bounds rect 
`CGRect boundsRect = CGRectStandardize(self.view.bounds);`.

**Review by codeman7 — APPROVED** [review]

**Comment by jverkoey** [comment]
What's the state of this PR? Either land or close please.

**Pull request merged**